### PR TITLE
Support Window menu

### PR DESCRIPTION
**Pull request by wineee**
DONE
- [x] WindowMenu for csd and ssd
- [x] better event Handler TitleBar.qml
- [x] Move to Work Space Left/Right
- [x] New/Remove Workspace in debug bar
- [x] Resize
- [x] Alway's on Top
- [x] Alway's on Visible Workspace

### CODE REVIEW

File: examples/tinywl-new/OutputMenuBar.qml [hunk 1]
```diff
@@ -102,6 +102,16 @@ Item {
                 }
             }
 
+            ToolButton {
+                text: "New Workspace"
+                onClicked: Helper.workspace.createContainer("Workspace"+Math.random());
```
Comment (zccrs) [inline examples/tinywl-new/OutputMenuBar.qml:107]: Why is random, should is a auto-increment number

Comment (wineee) [reply examples/tinywl-new/OutputMenuBar.qml:107]: It's random in kwin
```
Id_1=9aaa9c14-b594-49b4-b52a-b2b72fbad632
Id_2=a82c8ff4-20ca-4d96-bce3-2e6489b0a62b
Id_3=f00821b4-f311-4a13-90a0-ee5c39e3daab
Id_4=b2531c33-bf31-4338-a676-a3b28351d9d2
```
Keep auto-increment number should be meaningless，Aready have an `index`

File: examples/tinywl-new/workspacemodel.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+class SurfaceWrapper;
+class Workspace;
+class WorkspaceModel : public SurfaceListModel
+{
+    friend class Workspace;
+    Q_OBJECT
+    Q_PROPERTY(int index READ index NOTIFY indexChanged FINAL)
+    Q_PROPERTY(QString name READ name WRITE setName NOTIFY nameChanged FINAL)
+    Q_PROPERTY(bool visable READ visable WRITE setVisible NOTIFY visableChanged FINAL)
```
Comment (asterwyx) [inline examples/tinywl-new/workspacemodel.h:15]:
```suggestion
    Q_PROPERTY(bool visible READ visible WRITE setVisible NOTIFY visibleChanged FINAL)
```

File: examples/tinywl-new/workspacemodel.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Q_SIGNALS:
+    void nameChanged();
+    void indexChanged();
+    void visableChanged();
+
+private:
+    QString m_name;
+    int m_index = -1;
+    bool m_visable = false;
```
Comment (asterwyx) [inline examples/tinywl-new/workspacemodel.h:42]:
```suggestion
    bool m_visible = false;
```

**Pull request merged**